### ISSUE TITLE
Relevant search does not work

### ISSUE THREAD

**Issue by rkotov93** [ankane/searchkick#663]
Hello!

I have a problem with a relevant search. Results of following request are very strange:

``` ruby
Candidate.search('martin', fields: [:first_name, :last_name], match: :word_start, misspellings: false).map(&:name)


["Kautzer Martina",
 "Funk Martin",
 "Jaskolski Martin",
 "Gutmann Martine",
 "Wiegand Martina",
 "Schueller Martin",
 "Dooley Martin",
 "Stiedemann Martine",
 "Bartell Martina",
 "Gerlach Martine",
 "Green Martina",
 "Lang Martine",
 "Legros Martine",
 "Ernser Martina",
 "Boehm Martina",
 "Green Martine",
 "Nolan Martin",
 "Schmidt Martin",
 "Hoppe Martin",
 "Macejkovic Martine",
 "Emard Martine"]
```

**Why Martina is going earlier than Martin?!** 😟 

Searckick config:

``` ruby
searchkick language: %w(German English), word_start: [:first_name, :last_name]
```

P.S.
Also I tried to do that request

``` ruby
Candidate.search('martin', fields: ['first_name^10', 'last_name^10'], match: :word_start, misspellings: false)
```

but it didn't hep me.

**Comment by m2pomorski** [comment]
I have the same problem. Any solution?

**Comment by rkotov93** [comment]
@m2pomorski Unfortunately not. I've created a [question](http://stackoverflow.com/questions/36862825/searchkick-results-are-not-relevant) on stackoverflow, but didn't get any correct solution.

**Comment by ankane** [comment]
@rkotov93 Searchkick doesn't currently give more weight to exact matches with word start, but I'm happy to accept a PR for it.

**Comment by MK-42** [comment]
@ankane could you point me in direction where to implement that? I would like to have that for text_middle, maybe it's possible to resolve all of them in one go.

**Comment by ankane** [comment]
Hey @MK-42, thanks for the offer. This was fixed in `1.4.0`.
